### PR TITLE
sql: add support for DESC indexes using annotation and atlasgo.io

### PR DESCRIPTION
**Pull request by a8m**
Fixed https://github.com/ent/ent/issues/1631

### CODE REVIEW

File: dialect/entsql/annotation.go [hunk 1]
```diff
@@ -202,6 +202,29 @@ type IndexAnnotation struct {
...
+	//
+	//	index.Fields("c1", "c2", "c3").
+	//		Annotation(
+	//			entsql.DescColumn("c1"),
+	//			entsql.DescColumn("c2"),
+	//		)
+	//
+	//	CREATE INDEX `table_c1_c2_c3` ON `table`(`c1` DESC, `c2` DESC, `c3`)
+	//
+	DescColumns map[string]bool
```
Comment (rotemtam) [inline dialect/entsql/annotation.go:227]: isn't order lost by using a map?

**Review by rotemtam — COMMENTED** [review]
looks good, see question

File: dialect/entsql/annotation.go (hunk 1 above)
Comment (a8m) [reply dialect/entsql/annotation.go:227]: The order has no meaning here as I use `DescColumns[c.Name]` to check if a `DESC` option was defined (not looping over it).

File: dialect/entsql/annotation.go [hunk 2]
```diff
@@ -202,6 +202,28 @@ type IndexAnnotation struct {
...
+	// In MySQL, the following annotation maps to:
+	//
+	//	index.Fields("c1", "c2", "c3").
+	//		Annotation(
+	//			entsql.DescColumns("c1", "c2"),
+	//		)
+	//
+	//	CREATE INDEX `table_c1_c2_c3` ON `table`(`c1` DESC, `c2` DESC, `c3`)
+	//
+	DescColumns map[string]bool
```
Comment (masseelch) [inline dialect/entsql/annotation.go:226]: The more "goish" (and in my opinion cleaner way) is going with `map[string]struct{}`. Not really sure though, the check adds some boilerplate to it. 

Will this be used by the user directly? Looks like it is unlikely.

**Review by masseelch — APPROVED** [review]
Minor comment.

File: dialect/entsql/annotation.go (hunk 2 above)
Comment (a8m) [reply dialect/entsql/annotation.go:226]: You still have the option to add annotations like:
```go
&entsql.IndexAnnotation{
	Prefix: 10,
	Desc: true,
}
```

**Comment by a8m** [comment]
Thanks @rotemtam and @masseelch for the review ❤️

**Pull request merged**